unlock-app/src/pages/404.tsx

```diff
@@ -0,0 +1,19 @@
+import { NextPage } from 'next'
+import { AppLayout } from '~/components/interface/layouts/AppLayout'
+import { ImageBar } from '~/components/interface/locks/Manage/elements/ImageBar'
+
+const Page404: NextPage = () => {
```

Comment:
Will next.js handle showing it when the path does not match anything?

Reply:
For the direct hit, no. We do static export so the page will need to be handled by the netlify side on direct hits. 

For router navigation, nextjs should handle it.

Reply:
But when I go to `https://app.unlock-protocol.com/hahahahahaha`, it looks like next is able to catch that?

Reply:
it's not nextjs. That's a 404.html page generated by nextjs which gets served automatically by netlify since they handle 404s properly. 

If you were to change host to where it doesn't handle 404s like netlify, it won't work.

Reply:
https://docs.netlify.com/routing/redirects/redirect-options/#custom-404-page-handling